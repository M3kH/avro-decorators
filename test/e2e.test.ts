import { promises as fs } from 'fs'
import rimraf from 'rimraf'
import * as path from 'path'
import * as execa from 'execa'

describe('e2e', () => {

  beforeEach(async () => {
    await new Promise<void>((rs, rj) => rimraf(path.join(__dirname, '../schemas'), (err) => err ? rj() : rs()))
  })

  test(`run 'generate' for test-app-1 with config in root`, async () => {
    const command = 'node ../../build/bin/cli.js generate'
    const cwd = './test/test-app-1'

    const { stdout: output, stderr: errorOutput } = await execa.command(command, { cwd })

    expect(errorOutput).toEqual('')

    expect(output).toMatchSnapshot()
  })

  test(`run 'generate -c <config-path>' for test-app-1 with config in different folder`, async () => {
    const command =
      'node ../../build/bin/cli.js generate -c test-config-write-to-file.config.ts'
    const cwd = './test/test-app-1'

    const { stdout: output, stderr: errorOutput } = await execa.command(command, { cwd })

    const schema = (await fs.readFile(path.join(__dirname, 'test-app-1/schemas/Fruit.avsc'))).toString()

    expect(errorOutput).toEqual('')
    expect(output).toEqual('Writing 1 files...\n * wrote Fruit to Fruit.avsc')
    expect(schema).toMatchSnapshot()

  })
})
