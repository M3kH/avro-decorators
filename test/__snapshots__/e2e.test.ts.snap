// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e run 'generate -c <config-path>' for test-app-1 with config in different folder 1`] = `
"{
  \\"type\\": \\"record\\",
  \\"name\\": \\"Fruit\\",
  \\"fields\\": [
    {
      \\"name\\": \\"id\\",
      \\"type\\": \\"int\\"
    },
    {
      \\"name\\": \\"name\\",
      \\"type\\": \\"string\\",
      \\"doc\\": \\"The name of the fruit\\"
    },
    {
      \\"name\\": \\"origin\\",
      \\"type\\": {
        \\"type\\": \\"record\\",
        \\"name\\": \\"Origin\\",
        \\"fields\\": [
          {
            \\"name\\": \\"continent\\",
            \\"type\\": \\"string\\",
            \\"doc\\": \\"The continent the fruit is originally from\\"
          }
        ],
        \\"namespace\\": \\"fruits.meta\\",
        \\"doc\\": \\"An origin location\\"
      },
      \\"doc\\": \\"The origin of the fruit\\"
    },
    {
      \\"name\\": \\"fruitType\\",
      \\"type\\": {
        \\"type\\": \\"enum\\",
        \\"name\\": \\"FruitType\\",
        \\"symbols\\": [
          \\"berry\\",
          \\"tropical\\"
        ]
      }
    },
    {
      \\"name\\": \\"flavours\\",
      \\"type\\": {
        \\"type\\": \\"array\\",
        \\"items\\": \\"string\\"
      },
      \\"default\\": []
    },
    {
      \\"name\\": \\"inventory\\",
      \\"type\\": {
        \\"type\\": \\"map\\",
        \\"values\\": \\"int\\"
      }
    },
    {
      \\"name\\": \\"anything\\",
      \\"type\\": [
        \\"string\\",
        \\"Fruit\\",
        {
          \\"type\\": \\"enum\\",
          \\"name\\": \\"Letter\\",
          \\"symbols\\": [
            \\"A\\",
            \\"B\\",
            \\"C\\"
          ]
        },
        {
          \\"type\\": \\"array\\",
          \\"items\\": \\"string\\"
        },
        {
          \\"type\\": \\"map\\",
          \\"values\\": \\"string\\"
        },
        {
          \\"type\\": \\"fixed\\",
          \\"name\\": \\"MyFixedThing\\",
          \\"size\\": 4
        },
        {
          \\"type\\": \\"record\\",
          \\"name\\": \\"MyInlineRecord\\",
          \\"fields\\": []
        }
      ],
      \\"doc\\": \\"An examplary union\\"
    }
  ]
}
"
`;

exports[`e2e run 'generate' for test-app-1 with config in root 1`] = `
"Printing schemas to stdout, as option 'outDir' is not set in config

Fruit - Fruit.avsc

{
  \\"type\\": \\"record\\",
  \\"name\\": \\"Fruit\\",
  \\"fields\\": [
    {
      \\"name\\": \\"id\\",
      \\"type\\": \\"int\\"
    },
    {
      \\"name\\": \\"name\\",
      \\"type\\": \\"string\\",
      \\"doc\\": \\"The name of the fruit\\"
    },
    {
      \\"name\\": \\"origin\\",
      \\"type\\": {
        \\"type\\": \\"record\\",
        \\"name\\": \\"Origin\\",
        \\"fields\\": [
          {
            \\"name\\": \\"continent\\",
            \\"type\\": \\"string\\",
            \\"doc\\": \\"The continent the fruit is originally from\\"
          }
        ],
        \\"namespace\\": \\"fruits.meta\\",
        \\"doc\\": \\"An origin location\\"
      },
      \\"doc\\": \\"The origin of the fruit\\"
    },
    {
      \\"name\\": \\"fruitType\\",
      \\"type\\": {
        \\"type\\": \\"enum\\",
        \\"name\\": \\"FruitType\\",
        \\"symbols\\": [
          \\"berry\\",
          \\"tropical\\"
        ]
      }
    },
    {
      \\"name\\": \\"flavours\\",
      \\"type\\": {
        \\"type\\": \\"array\\",
        \\"items\\": \\"string\\"
      },
      \\"default\\": []
    },
    {
      \\"name\\": \\"inventory\\",
      \\"type\\": {
        \\"type\\": \\"map\\",
        \\"values\\": \\"int\\"
      }
    },
    {
      \\"name\\": \\"anything\\",
      \\"type\\": [
        \\"string\\",
        \\"Fruit\\",
        {
          \\"type\\": \\"enum\\",
          \\"name\\": \\"Letter\\",
          \\"symbols\\": [
            \\"A\\",
            \\"B\\",
            \\"C\\"
          ]
        },
        {
          \\"type\\": \\"array\\",
          \\"items\\": \\"string\\"
        },
        {
          \\"type\\": \\"map\\",
          \\"values\\": \\"string\\"
        },
        {
          \\"type\\": \\"fixed\\",
          \\"name\\": \\"MyFixedThing\\",
          \\"size\\": 4
        },
        {
          \\"type\\": \\"record\\",
          \\"name\\": \\"MyInlineRecord\\",
          \\"fields\\": []
        }
      ],
      \\"doc\\": \\"An examplary union\\"
    }
  ]
}


"
`;
